### ISSUE TITLE
Custom Block, theme.json definitions not loaded with latest RC Gutenberg Plugin

### ISSUE THREAD

**Issue by SGr33n** [WordPress/gutenberg#43679]
### Description

Hi,
I'm just trying the latest Gutenberg RC and I can see that something is not working anymore.
The issue is that I created a custom excerpt block, called theme/single-post-excerpt and I added some css to the theme.json:

`			"theme/single-post-excerpt": {
				"typography": {
					"lineHeight": "var(--wp--custom--typography--line-height--small)",
					"fontSize": "var(--wp--preset--font-size--large)",
					"fontWeight": "500"
				}
			}`

That definitions works good on the latest WP 6.0.1 but if I install the latest Gutenberg RC they are ignored.

### Step-by-step reproduction instructions

Create a custom block.
Add css definitions for that block on the theme.json
Check if they are applied.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by talldan** [comment]
Hi @SGr33n, would you be able to provide some more detailed steps for reproduction. Either fuller code snippets that a developer can paste, or a link to another repo/plugin that has the problem. Thanks.

**Comment by SGr33n** [comment]
Hi @talldan!
Sure, how can I send you my code?
Or I could enable you on a staging environment with this issue already on.

**Comment by t-hamano** [comment]
I was able to reproduce it.

<details>
<summary>Test Code</summary>

`plugins/my-custom-block/src/index.js`

```javascript
import { registerBlockType } from '@wordpress/blocks';
import { useBlockProps } from '@wordpress/block-editor';
import metadata from './block.json';

registerBlockType( metadata.name, {
	edit: () => <p { ...useBlockProps() }>My Custom  Block</p>,
	save: () => <p { ...useBlockProps.save() }>My Custom Block</p>,
} );
```

`plugins/my-custom-block/src/block.json`

```json
{
	"apiVersion": 2,
	"name": "create-block/test-block",
	"title": "My Custom Block",
	"category": "text",
	"textdomain": "test-block",
	"editorScript": "file:./index.js"
}
```

`themes/twenty-twenty-two/theme.json`

```json
{
	"version": 2,
	"styles": {
		"blocks": {
			"core/heading": {
				"border": {
					"width": "3px",
					"color": "#f00",
					"style": "solid"
				},
				"color": {
					"text": "#fff",
					"background": "#000"
				}
			},
			"create-block/test-block": {
				"border": {
					"width": "3px",
					"color": "#f00",
					"style": "solid"
				},
				"color": {
					"text": "#fff",
					"background": "#000"
				}
			}
		}
	}
}
```
</details>

My test results show that when I activate the latest Gutenberg, the styles defined in `theme.json` aren't applied to the custom block only on the front-end.


https://user-images.githubusercontent.com/54422211/189512272-89a97534-8307-45ce-a9aa-099a84a6654f.mp4

**Comment by SGr33n** [comment]
@t-hamano exactly what happened to me.

**Comment by t-hamano** [comment]
@SGr33n 
Thank you for the reply.
I believe #44053 will fix this issue.

**Comment by t-hamano** [comment]
On behalf of #44053, #44018 will fix this issue.

**Comment by t-hamano** [comment]
@SGr33n 
This issue should have been resolved by #44018, so I would like to close this issue.
If you have any issues, please feel free to reopen.

**Comment by SGr33n** [comment]
Sure, thanks @t-hamano !
